File: Python/Szyfr.py
n = int(input())
for _ in range(n):
    decoded = []
    binary = input().strip()
    groups = [binary[i:i+5] for i in range(0, len(binary), 5) if len(binary[i:i+5]) == 5]
    for group in groups:
        num = int(group, 2)
        decoded_char = chr(num % 26 + ord('A'))
        decoded.append(decoded_char)
    print(''.join(decoded))

File: Java/Szyfr.java
import java.util.Scanner;

public class Main
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        input.nextLine();
        for (int i = 0; i < n; i++) 
        {
            String binary = input.nextLine();
            StringBuilder decoded = new StringBuilder();
            for (int j = 0; j + 4 < binary.length(); j += 5) 
            {
                String group = binary.substring(j, j + 5);
                int num = Integer.parseInt(group, 2);
                char decodedChar = (char) ('A' + (num % 26));
                decoded.append(decodedChar);
            }
            System.out.println(decoded.toString());
        }
        input.close();
    }
}